osu.Game/Screens/OsuScreen.cs

```diff
@@ -44,9 +44,11 @@ public abstract class OsuScreen : Screen, IOsuScreen, IHasDescription
         public virtual bool HideOverlaysOnEnter => false;
 
         /// <summary>
-        /// Whether overlays should be able to be opened once this screen is entered or resumed.
+        /// The initial initial overlay activation mode to use when this screen is entered for the first time.
```

Comment:
Doesn't sound clear enough (also double word typo, I just realized it as I was writing the suggestion), how about:
```suggestion
        /// The initial overlay activation mode, this would be set to <see cref="OverlayActivationMode"/> on constructor.
```

Reply:
original is better if the double word is removed

Reply:
what if the overlay activation mode has been changed before the screen is entered, I've suggested it to redirect to the `OverlayActivationMode` bindable which is the one its value will be used when the screen is entered.

Reply:
actually might have been too picky about it, I just didn't like the fact that it explicitly says "it will be used when the screen is entered for the first time", maybe at least appending `, unless OverlayActivationMode changed prior to that.` to it.